sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java

```diff
@@ -69,7 +69,8 @@ public String getToken() {
 
     private static Long parseDateToEpochSeconds(String dateTime) {
         try {
-            return Long.parseLong(dateTime);
+            Long expiresInSeconds = Long.parseLong(dateTime);
+            return Instant.from(OffsetDateTime.now().plusSeconds(expiresInSeconds).toInstant()).getEpochSecond();
```

Comment:
```suggestion
            return OffsetDateTime.now().plusSeconds(expiresInSeconds).toEpochSecond();
```

I think since we already are using `OffsetDateTime.now()` we don't need to go through `Instant` for this, unlike the other cases where we get a `TemporalAccessor`.